lib/rules/no-return-await.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+const functionTypes = new Set([
+    "FunctionExpression",
+    "FunctionDeclaration",
+    "ArrowFunctionExpression",
+    "MethodDefinition",
+]);
+
+/**
+ * @param {ASTNode} node the node from which to locate a containing function
+ * @returns {ASTNode} the containing function node
+ */
+function getContainingFunction(node) {
+    if (!node.parent || functionTypes.has(node.type)) {
```

Comment:
I don't think this function is necessary anymore (see my comment below), but if we do end up keeping it, it might be better to use [`astUtils.isFunction`](https://github.com/eslint/eslint/blob/e117b80cf2ed7b712057d3c2194c1c626edd3ce3/lib/ast-utils.js#L611-L613) to check whether a node is a function.